core/trino-main/src/main/java/io/trino/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -393,9 +393,11 @@ protected Boolean visitFunctionCall(FunctionCall node, Void context)
                                     .map(NodeRef::of)
                                     .map(columnReferences::get)
                                     .filter(Objects::nonNull)
+                                    .map(ResolvedField::getFieldId)
                                     .collect(toImmutableList());
                             for (Expression sortKey : sortKeys) {
-                                if (!node.getArguments().contains(sortKey) && !fieldIds.contains(columnReferences.get(NodeRef.of(sortKey)))) {
+                                ResolvedField field = columnReferences.get(NodeRef.of(sortKey));
```

Comment:
Now instead of `X.contains(map.get(A))` we need to do
```
A a = map.getA();
return ( a == null ) || !X.contains(a.getB());
```
since the value type of the map changed.
But you are right, I replaced that with a double if